File: Julia/[0] relativity in Julia/scenarios/lambda_L_L_M/variables.jl
module Variables

using CSV
using DataFrames

# Export the symbols needed in other modules.
export lambda_staro, R_star, beta, this_dir, data_cond_path, data_gas_path,
       data_cond, data_gas, x0, n0, n10, m0, m10, R0, DR0, Ms0, delta,
       rho, nHS, mHS, f1hoy, c2, c1, P0, Mb0, G, c, M_sol, rho_nuc, rho_c, xrho,
       Po, T, r0, f0, f10, f20, f30, test_variable

# --------------------------
# Model Parameters and Paths
# --------------------------
const lambda_staro = 1.56
const R_star = 1.0
const beta = 1

# the absolute path to this script's directory
const this_dir = @__DIR__

# Build paths to the data files (portable)
const data_cond_path = joinpath(this_dir, "data", "init_cond.dat")
const data_gas_path  = joinpath(this_dir, "data", "init_cond.dat")  # Adjust if file name differs

# --------------------------
# Read Data Files into DataFrames
# --------------------------
const data_cond = CSV.File(data_cond_path) |> DataFrame
const data_gas  = CSV.File(data_gas_path) |> DataFrame

# --------------------------
# Assign Variables from "data_cond"
# --------------------------
const x0 = data_cond[1, 1]
const n0 = data_cond[2, 1]
const n10 = data_cond[3, 1]
const m0 = data_cond[4, 1]
const m10 = data_cond[5, 1]
const R0 = data_cond[6, 1]   # This is the value we want to find via a shooting method
const DR0 = data_cond[7, 1]
const Ms0 = data_cond[8, 1]
const delta = data_cond[9, 1]

# --------------------------
# Assign Variables from "data_gas"
# --------------------------
const rho = data_gas[1, 1]
const nHS = data_gas[2, 1]
const mHS = data_gas[3, 1]  # After squaring, used in f2(R)
const f1hoy = data_gas[4, 1]

# --------------------------
# Define Other Variables
# --------------------------
const c2 = (-1 * nHS * 19 * 41^(-1 * nHS - 1)) / f1hoy
const c1 = 19 * c2
const P0 = 0.1 * rho

const Mb0 = 1.66e-27   # initial baryonic mass?
const G = 1
const c = 1
const M_sol = 1

const rho_nuc = 1.66e17
const rho_c = (c^8) / (M_sol^2 * G^3)
const xrho = (rho_nuc * c^2) / rho_c

const Po = 0.1 * rho
const T = -rho + 3 * Po

# --------------------------
# Initial Conditions Vector
# --------------------------
const r0 = [n0, m0, R0, DR0, P0, Ms0, Mb0]

# --------------------------
# Define Functions for Numerical Method Comparison
# --------------------------
const f0 = R0 - (c1 * mHS^2 * (R0 / mHS^2)^nHS) / (1 + c2 * (R0 / mHS^2)^nHS)
const f10 = 1 - (c1 * nHS * (R0 / mHS^2)^(-1 + nHS)) / ((1 + c2 * (R0 / mHS^2)^nHS)^2)
const f20 = (c1 * mHS^2 * nHS * (R0 / mHS^2)^nHS * (1 + c2 * (R0 / mHS^2)^nHS + nHS * (-1 + c2 * (R0 / mHS^2)^nHS))) /
            (R0^2 * (1 + c2 * (R0 / mHS^2)^nHS)^3)
const f30 = -(c1 * mHS^2 * nHS * (R0 / mHS^2)^nHS * (2 * (1 + c2 * (R0 / mHS^2)^nHS)^2 +
            3 * nHS * (-1 + c2^2 * (R0 / mHS^2)^(2 * nHS)) +
            nHS^2 * (1 - 4 * c2 * (R0 / mHS^2)^nHS + c2^2 * (R0 / mHS^2)^(2 * nHS)))) /
            (R0^3 * (1 + c2 * (R0 / mHS^2)^nHS)^4)

# Test variable to check proper import.
const test_variable = "we got the information from the original file! after nesting these files :)"

end  # module Variables

File: Python/Relativity_Modules_SSS_mods/Relativity_Modules_SSS_mod6.0_just__solve_ivp/scenarios/alpha_L_L_M/variables.py
import numpy as np
import pandas as pd
import os

# parameters for this "f(R)" function
alpha_mir = 1.2
R_star = 1


# the absolute path to the directory containing this file is
this_dir = os.path.dirname(__file__)

# the path to the data files will be defined by
data_cond_path = os.path.join(this_dir, "data", "init_cond.dat")
data_gas_path  = os.path.join(this_dir, "data", "init_gas.dat")

data_cond = pd.read_csv(data_cond_path, sep=",")
data_gas  = pd.read_csv(data_gas_path,  sep=",")


# print(data_cond)
# print(data_gas)
# print(type(data_gas.values[1][0]))
# print(data_gas.values[1][0])

# it works, now lets assign variables from "data_cond"

x0 = data_cond.values[0][0]

n0 = data_cond.values[1][0]

n10 = data_cond.values[2][0]

m0 = data_cond.values[3][0]

m10 = data_cond.values[4][0]

R0 = data_cond.values[5][0]  # this is the value we want to adjust as the program runs (when it runs correctly)

DR0 = data_cond.values[6][0]

Ms0 = data_cond.values[7][0]

delta = data_cond.values[8][0]

# assign variables from "data_gas"

rho = data_gas.values[0][0]

nHS = data_gas.values[1][0]

mHS = data_gas.values[2][0]  # the article says that after the square of this value, "R" values input into what we'll define as "f2(R)", are positive.

f1hoy = data_gas.values[3][0]

# TODO 3: define the other ("non-imported") variables used throughout the program
c2 = np.divide(np.multiply(-1, nHS * 19 * np.power(41, -1 * nHS - 1)), f1hoy)

c1 = np.multiply(19, c2)

P0 = np.multiply(0.1, rho)

Mb0 = 1.66e-27  # is this the initial baryonic mass?

G = 1

c = 1

M_sol = 1  # en el archivo original sale como "M_sol  ==> 1.0"

rho_nuc = 1.66e17

rho_c = np.divide(np.power(c, 8), np.multiply(np.power(M_sol, 2), np.power(G, 3)))

xrho = np.divide(np.multiply(rho_nuc, np.power(c, 2)), rho_c)

Po = np.multiply(0.1, rho)

T = np.add(np.multiply(-1, rho), np.multiply(3, Po))

# TODO 3.1:  the initial conditions will be stored in
r0 = np.array([n0, m0, R0, DR0, P0, Ms0, Mb0], float)

# other variables defined to compare the numerical method with an analytic solution for "R"

f0 = R0 - (c1 * mHS ** 2 * (R0 / mHS ** 2) ** nHS) / (1 + c2 * (R0 / mHS ** 2) ** nHS)

f10 = 1 - (c1 * nHS * (R0 / mHS ** 2) ** (-1 + nHS)) / ((1 + c2 * (R0 / mHS ** 2) ** nHS) ** 2)

f20 = (c1 * mHS ** 2 * nHS * (R0 / mHS ** 2) ** nHS * (
            1 + c2 * (R0 / mHS ** 2) ** nHS + nHS * (-1 + c2 * (R0 / mHS ** 2) ** nHS))) / (
                  R0 ** 2 * (1 + c2 * (R0 / mHS ** 2) ** nHS) ** 3)

f30 = -(c1 * mHS ** 2 * nHS * (R0 / mHS ** 2) ** nHS * (2 * (1 + c2 * (R0 / mHS ** 2) ** nHS) ** 2 + 3 * nHS * (
            -1 + c2 ** 2 * (R0 / mHS ** 2) ** (2 * nHS)) + nHS ** 2 * (1 - 4 * c2 * (
            R0 / mHS ** 2) ** nHS + c2 ** 2 * (R0 / mHS ** 2) ** (2 * nHS)))) / (
                  R0 ** 3 * (1 + c2 * (R0 / mHS ** 2) ** nHS) ** 4)
